### PR TITLE
WT-13611 Move log's header files into src/log

### PR DESCRIPTION
**Pull request by ajmorton**
Move the header files relevant to log/ from `src/include` into `src/log`. This consolidates all of our log code into a single location.

To do so:
1. Add `log` to `SELF_CONTAINED_MODULES` in `prototypes.py` and run s_all
2. Move the contents of `src/include/log.h` into the public and private headers. The `tools/modularity_check/modularity_check.py` tools can be used to advise what data is public and private. See the modularity playbook for how to use it. (This tool will be delivered in WT-13613)

### CODE REVIEW

File: test/model/test/common/wiredtiger_util.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include "wiredtiger.h"
 extern "C" {
 #include "test_util.h"
+#include "../../../../src/log/log_private.h"
```
Comment (ajmorton) [inline test/model/test/common/wiredtiger_util.cpp:35]: This is ugly but necessary for the moment. It can be cleaned up as part of WT-13610

Comment (bvpvamsikrishna) [reply test/model/test/common/wiredtiger_util.cpp:35]: sgtm

File: src/log/log_private.h [hunk 2]
```diff
@@ -312,42 +277,6 @@ struct __wt_log {
...
-    record->checksum = __wt_bswap32(record->checksum);
-    record->flags = __wt_bswap16(record->flags);
-    record->mem_len = __wt_bswap32(record->mem_len);
-#else
-    WT_UNUSED(record);
-#endif
-}
-
 /*
  * WT_LOG_DESC --
```
Comment (bvpvamsikrishna) [inline src/log/log_private.h:281]: currently __wt_log_desc is being used in [tools/hexfiend/Templates/WT/WT-Log.tcl](https://github.com/wiredtiger/wiredtiger/blob/ba14e16959924a2a970998878e104a04bb79587b/tools/hexfiend/Templates/WT/WT-Log.tcl#L35)

I am not sure about the usage of this file, if it is required, may be we need keep it in log.h

Comment (ajmorton) [reply src/log/log_private.h:281]: I ran hexfiend with the WT_Log template to confirm, and there's no issues here. 
Attached below is a screenshot of hexfiend decoding the LOG_DESC field correctly (see the blue highlighted section on the right). 
Screenshot 2024-10-14 at 10 36 20 AM

Hexfiend templates are an independent definition of the on-disk data layout. This should be stable given it's on disk, but it's a good callout that hexfiend has a dependency on on-disk data formatting.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**